src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -181,6 +187,7 @@ static SerializationManager()
             AppDomain.CurrentDomain.AssemblyResolve += OnResolveEventHandler;
 
             registeredTypes = new HashSet<Type>();
+            externalSerializers = new ConcurrentHashSet<IExternalSerializer>();
```

Comment:
Good point.  It is using ConcurrentHashSet<T> because the code was, at one point, dynamically discovering any IExternalSerializers when assemblies were loaded which could result in concurrency issues.